commons/src/main/java/com/powsybl/commons/datasource/DataSourceUtil.java

```diff
@@ -63,7 +64,7 @@ static DataSource createDataSource(Path directory, String fileNameOrBaseName, Da
         Objects.requireNonNull(fileNameOrBaseName);
 
         if (fileNameOrBaseName.endsWith(".zip")) {
-            return new ZipFileDataSource(directory, getBaseName(fileNameOrBaseName.substring(0, fileNameOrBaseName.length() - 4)), observer);
+            return new ZipFileDataSource(directory, fileNameOrBaseName.substring(0, fileNameOrBaseName.length() - 4), observer);
```

Comment:
If we do it for other data sources, we will break some behaviours when there are multiple files.

For example, if we have an anonymization dictionary for IIDM:

- `network.xiidm` will be found, OK
- `network_mapping.csv` will not be found, because expected filename will be `network.xiidm_mapping.csvg`

So this workaround can only work for zip files containing only one file.

Reply:
Ok, got it, thanks for the example, we'd better rename this PR if we keep it that way. So (trying to sum up things for myself!) this only fixes *zip* data sources with dot in name, and only if the zip file 
- contains several files but the zip file name does not contain the extension (for instance `network.zip` containing `network.xiidm` and `network_mapping.csv`)
- or contains only one file but has a name containing the extension (`network.xiidm.zip` containing `network.xiidm` file)

The importer is still not working with a `network_v1.0.xiidm` file, or with a `network_v1.0.xiidm.gz` file.

Is there any reason to take the first dot (and not the last) for a non-compressed file by the way?